numpy/core/_add_newdocs.py

```diff
@@ -10,9 +10,16 @@
 """
 from __future__ import division, absolute_import, print_function
 
+import sys
+
 from numpy.core import numerictypes as _numerictypes
 from numpy.core import dtype
-from numpy.core.function_base import add_newdoc
+from numpy.core.function_base import add_newdoc as _add_newdoc
```

Comment:
Rather than changing the name on import, could you change `function_base.add_newdoc` to `function_base._add_newdoc`

Reply:
`np.add_newdoc` is public API, I think this will break that.

Reply:
I see. What a mess: there is `numpy.core.function_base` and `numpy.lib.function_base`, each adds different things to `__all__`. 

Since the implementation is specific to the `_multiarray_umath` module, why should `add_newdoc`  be part of the public API? Its only use is at startup to modify the uninitialized `tp_doc` slot in `_multiarray_umath` methods.

Reply:
... but not really relevant to this PR

Reply:
I don't think `np.add_newdoc` should be public, but as a bugfix this PR tries not to break the status quo and leaves it where it is.

Reply:
Either way, evidently changing `add_newdocs` is a can of worms, so let's not mix it with the bug fixes in this patch.